### PR TITLE
fix(deploy-tooling): handle 401 error for cloud target systems

### PR DESCRIPTION
**Pull request by cianmSAP**
#1131

When a 401 error is thrown in deployment, the user is prompted for credentials. 
In the case where a serviceKey is used, the uaa params need to be updated

### CODE REVIEW

File: packages/deploy-tooling/test/unit/base/deploy.test.ts [hunk 1]
```diff
@@ -200,6 +200,31 @@ describe('base/deploy', () => {
...
+                        clientsecret: '~clientsecret',
+                        url: '~url'
+                    }
+                },
+                url: '~targetUrl'
+            } as AbapTarget;
+            mockedUi5RepoService.deploy.mockRejectedValueOnce(axiosError(401));
+            await deploy(archive, { app, target: cloudTarget, yes: true }, nullLogger);
+            prompts.inject(['~uaa-username', '~uaa-password']);
+            expect(mockedUi5RepoService.deploy).toBeCalledWith({
```
Comment (longieirl) [inline packages/deploy-tooling/test/unit/base/deploy.test.ts:220]: Wondering if we should be more explicit what is actually changed here i.e. the serviceKey.uaa object with the updated values. Its not clear from the test?

Comment (cianmSAP) [reply packages/deploy-tooling/test/unit/base/deploy.test.ts:237]: Good point. I've updated the test

Comment (longieirl) [reply packages/deploy-tooling/test/unit/base/deploy.test.ts:235]: Thank you.

**Review by longieirl — APPROVED** [review]

**Review by Klaus-Keller — APPROVED** [review]
- code changes are straight forward and easy to understand
- Test coverage is excellent
- Did not locally build and test, just visual review

Approved from my side.

**Pull request merged**